packages/flutter/lib/src/widgets/widget_state.dart

```diff
@@ -594,6 +780,95 @@ class _WidgetStatePropertyWith<T> implements WidgetStateProperty<T> {
   T resolve(Set<WidgetState> states) => _resolve(states);
 }
 
+/// A [Map] used to resolve to a single value of type `T` based on
+/// the current set of Widget states.
+///
+/// {@template flutter.widgets.WidgetStateProperty.WidgetStateMap}
+/// Example:
+///
+/// ```dart
+/// WidgetStateProperty<Color?>.fromMap(<WidgetStatesConstraint, Color?>{
```

Comment:
Maybe leave a comment on here to explain that the map passed to fromMap is of type `WidgetStateMap<Color?>` for people wondering why this example is given on WidgetStateMap.

I agree with your comment above that assigning it to a variable is not a good idea.